File: Sortings/insertion_sort.py
def insertion_sort(lst):
    for i in range(len(lst)):
        j = i
        while j>0 and lst[j] < lst[j-1]:
            lst[j], lst[j-1] = lst[j-1], lst[j]
            j -= 1

if __name__ == "__main__":
    lst = [9, 7, 5, 6, 2, 1, 4, 3, 0, 8]
    print("Unsorted list")
    print(lst)
    print("Sorted list")
    insertion_sort(lst)
    print(lst)

File: Sortings/insertionSort.cpp
#include<bits/stdc++.h>
using namespace std;

void swap(int& a, int& b) {
    int aux = a;
    a = b;
    b = aux;
}

void insertionSort(int arr[], int size) {
    for(int i=0; i<size; ++i) {
        for(int j=i; j>0 && arr[j] < arr[j-1]; --j) {
            swap(arr[j], arr[j-1]);
        }
    }
}

void printArr(int arr[], int size) {
    for(int i=0; i<size; ++i)
        cout << arr[i] << " ";
    cout << "\n";
}

int main() {
    int size = 10;
    int arr[10] = {9, 7, 5, 6, 2, 1, 4, 3, 0, 8};

    cout << "Unsorted array" << endl;
    printArr(arr, size);
    cout << "Sorted array" << endl;
    insertionSort(arr, size);
    printArr(arr, size);

    return 0;
}
